### PR TITLE
Refactors stacks to be FULLY ECS.

### PR DESCRIPTION
**Pull request by Zumorica**
Behavior still the same, but it's now more ECS-y.

~~Requires https://github.com/space-wizards/RobustToolbox/pull/1761 and submodule update.~~

### CODE REVIEW

File: Content.Shared/GameObjects/Components/SharedStackComponent.cs [hunk 1]
```diff
@@ -27,16 +21,25 @@ public abstract class SharedStackComponent : Component, ISerializationHooks
...
+        [DataField("stackType", required:true, customTypeSerializer:typeof(PrototypeIdSerializer<StackPrototype>))]
+        public string StackTypeId { get; private set; } = string.Empty;
+
+        [ViewVariables(VVAccess.ReadWrite)]
+        public int Count
         {
             get => _count;
             set
             {
+                if (_count == value)
```
Comment (ShadowCommander) [inline Content.Shared/GameObjects/Components/SharedStackComponent.cs:33]: Next step is to replace this with a ChangeStackEvent right?

**Review by ShadowCommander — APPROVED** [review]

File: Content.Shared/GameObjects/Components/SharedStackComponent.cs (hunk 1 above)
Comment (Zumorica) [reply Content.Shared/GameObjects/Components/SharedStackComponent.cs:33]: basically, yeah

Comment (Zumorica) [reply Content.Shared/GameObjects/Components/SharedStackComponent.cs]: done now!

**Comment by Zumorica** [comment]
The only thing left now would be removing the MaxCount logic from the component, but honestly since the current deisgn is so... weird I decided it wasn't worth the trouble

**Pull request merged**